### PR TITLE
Moves qtauto from recipe to class

### PR DESCRIPTION
**Pull request by tobsan**
This will make it easier to inherit for whoever wants to extend it.

Signed-off-by: Tobias Olausson <tobias.olausson@pelagicore.com>

### CODE REVIEW

File: classes/core-image-pelux-qtauto.bbclass [hunk 1]
```diff
@@ -0,0 +1,21 @@
...
+DESCRIPTION = "Class for inheriting a Pelux image with Qt frontend"
+
+inherit core-image-pelux
+
+# A bunch of QtAuto components
+QTAUTO_COMPONENTS := " \
+    qtapplicationmanager \
+    qtivi \
+    qtvirtualkeyboard \
+    dbus-session \
```
Comment (GordanM) [inline classes/core-image-pelux-qtauto.bbclass]: Maybe this should not be stated here explicitly but pulled in as a part of `qtappman.bbappend` dependency.

File: layers/b2qt/recipes-core/images/core-image-pelux-qtauto-neptune.bb [hunk 2]
```diff
@@ -0,0 +1,11 @@
+#
+#   Copyright (C) 2017 Pelagicore AB
+#   SPDX-License-Identifier: MIT
+#
+
+DESCRIPTION = "Reference PELUX image with QtAuto frontend"
+
+inherit core-image-pelux-qtauto
+
+# We use neptune as the reference UI
```
Comment (GordanM) [inline layers/b2qt/recipes-core/images/core-image-pelux-qtauto-neptune.bb]: Rename "We" to a more generic name, e.g. the image.

File: classes/core-image-pelux-qtauto.bbclass [hunk 3]
```diff
@@ -0,0 +1,21 @@
...
+"
+
+# Classes are special so we only do this if b2qt (which contains the components) is enabled
```
Comment (GordanM) [inline classes/core-image-pelux-qtauto.bbclass]: Maybe be more explicit and say: the b2qt meta layer is used/enabled, b2qt might not be obvious what it is to others.

File: classes/core-image-pelux-qtauto.bbclass [hunk 4]
```diff
@@ -0,0 +1,21 @@
...
+#   Copyright (C) 2017 Pelagicore AB
+#   SPDX-License-Identifier: MIT
+#
+
+DESCRIPTION = "Class for inheriting a Pelux image with Qt frontend"
+
+inherit core-image-pelux
+
+# A bunch of QtAuto components
+QTAUTO_COMPONENTS := " \
```
Comment (jonte) [inline classes/core-image-pelux-qtauto.bbclass]: `:=` is not necessary I think.. `=` should work, since there are no variables to expand?

Comment (tobsan) [reply classes/core-image-pelux-qtauto.bbclass]: Right, `:=` gives immediate expansion, but it is not necessary here. I was poking around with it earlier when it didn't work, but forgot to revert.

File: classes/core-image-pelux-qtauto.bbclass (hunk 1 above)
Comment (tobsan) [reply classes/core-image-pelux-qtauto.bbclass]: You mean only dbus-session, right?

Comment (GordanM) [reply classes/core-image-pelux-qtauto.bbclass]: That's correct, sorry for not being explicit.

**Pull request merged**